app/services/rapid_ready_for_decision/fast_track_pdf_generator.rb

```diff
@@ -19,22 +19,20 @@ class FastTrackPdfGenerator
       }
     }.freeze
 
-    def initialize(patient_info, blood_pressure_data, medications)
+    def initialize(patient_info, assessed_data, disability_type)
       @pdf = Prawn::Document.new
       @patient_info = patient_info
-      @blood_pressure_data = blood_pressure_data
-      @medications = medications
+      @blood_pressure_data = assessed_data[:bp_readings]
+      @medications = assessed_data[:medications]
       @date = Time.zone.today
+      @disability_type = disability_type
       @pdf.markup_options = PDF_MARKUP_SETTINGS
     end
 
     # progressively builds a pdf and is sensitive to sequence
     def generate
-      add_intro
-      add_blood_pressure_list
-      add_blood_pressure_outro
-      add_medications_list
-      add_about
+      template = File.join('app/services/rapid_ready_for_decision/views', "#{@disability_type}.erb")
```

Comment:
I like the idea of passing in the disability type (which I think is happening with having it in initialization and binding), and pointing towards a more generic base template that could use the disability type in logic for showing partials. I think something like this would be easier to scale than having a separate erb template per disability type.

Reply:
@leikkisa  I think I like that, it would look something like this:

```erb
<%# views/base.erb %>

<% if test ==:hypertension  %>
  <%= render_partial 'shared/intro' %>
  <%= render_partial 'hypertension/blood_pressure_readings' %>
  <%= render_partial 'hypertension/rating_schedule' %>
  <%= render_partial 'shared/medications' %>
  <%= render_partial 'hypertension/about' %>
<% end %>

<% if test ==:asthma  %>
  <%= render_partial 'shared/intro' %>
  <%= render_partial 'asthma/rating_schedule' %>
  <%= render_partial 'shared/medications' %>
  <%= render_partial 'asthma/about' %>
<% end %>
```

What do you think? Is this what you were thinking of?